File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number = get_long("Input your credit card number: ");
    int checksum = 0;
    int checkTwo;
    for (int i = 0; i < 16; i++)
    {
        if (number < 1000)
        {
            checkTwo = number;
        }
        // loop through the number and add
        if (i % 2 == 1)
        {
            checksum += (2 * (number % 10)) % 10 + (2 * (number % 10)) / 10;
        }
        else
        {
            checksum += number % 10;
        }
        number /= 10;
    }

    if (checksum % 10 == 0)
    {
        // getting first or first 2 digits
        printf("%i", checkTwo);
        // checking if first 2 digits satisfy condition
        if (((int) (checkTwo / 10)) == 4)
        {
            printf("VISA\n");
        }
        else if (checkTwo == 34 || checkTwo == 37)
        {
            printf("AMEX\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: pset6/sentimental-credit/credit.py
# TODO
number = 0

while number < 1:
    number = input("number: ")
    try:
        number = int(number)
    except:
        number = 0
number = str(number)

checksum = 0
for i in range(-1, -len(number)-1, -1):
    if i % 2 == 0:
        checksum += 2 * int(number[i])
    else:
        checksum += int(number[i])

if (checksum % 10 == 0) && (len(number) > )